src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -151,11 +151,11 @@ def _get_sku_object(cmd, sku):
     return sku
 
 
-def _grant_access(cmd, resource_group_name, name, duration_in_seconds, is_disk):
+def _grant_access(cmd, resource_group_name, name, duration_in_seconds, is_disk, access_level):
     AccessLevel = cmd.get_models('AccessLevel')
     client = _compute_client_factory(cmd.cli_ctx)
     op = client.disks if is_disk else client.snapshots
-    return op.grant_access(resource_group_name, name, AccessLevel.read, duration_in_seconds)
+    return op.grant_access(resource_group_name, name, access_level or AccessLevel.read, duration_in_seconds)
```

Comment:
i will add example commands to show the whole e2e including the storage commands